R/dl_build_bin.R

```diff
@@ -38,32 +39,39 @@ dl_build_bin <- function(sync=FALSE, GRAN.dir = './GRAN'){
 		stop('unrecognized OS type', os)
 	}
 	
-	gran_packages = available.packages(paste0('file:', src_dir), type='source')
+	#gran_packages = available.packages(paste0('file:', src_dir), type='source')
+	colnames(packages) <- c("Package","Version")
 	
+	#kill old versions or create directory
+	#unlink(file.path(build_dir), recursive=TRUE)
+	#dir.create(build_dir, recursive=TRUE)
+	toDelete <-  sub(".*\\/","",packages$package)
+	if(dir.exists(file.path(build_dir))){
+	  system(paste("rm",paste0(paste0(build_dir,"/",toDelete,"*"),collapse=" "))) #delete any existing version of toDelete packages
```

Comment:
It is generally a good rule of thumb to avoid `system()` as it can have all sorts of portability problems